modules/reindex/src/main/java/org/elasticsearch/index/reindex/ReindexTaskStateUpdater.java

```diff
@@ -122,6 +125,24 @@ public void onFailure(Exception ex) {
         });
     }
 
+    private ElasticsearchException assignmentFailureReason(ReindexTaskStateDoc oldDoc) {
+        if (oldDoc.isResilient()) {
+            if (oldDoc.getAllocationId() == null || allocationId > oldDoc.getAllocationId()) {
+                return null;
+            } else {
+                return new ElasticsearchException("A newer task has already been allocated");
+            }
+        } else {
+            if (oldDoc.getAllocationId() == null) {
+                return null;
+            } else {
+                return new ElasticsearchException("A prior task has already been allocated and reindexing is configured to be " +
```

Comment:
I think we should add a reference to the cluster level setting, mentioning its name here. This could nudge some users into trying it out.

If you defer adding the cluster level setting to a follow-up, let us add a todo here so that we remember it.